### ISSUE TITLE
Add a progress indicator on model inputs launch?

### ISSUE THREAD

**Issue by phargogh** [natcap/invest-workbench#111]
* Mac OS X 11
* Using mac artifact at https://pipelines.actions.githubusercontent.com/F4lHndai5pNrgEJlI2njzHq5AUtQLlP3jKt2FbnX1yv7QGCERr/_apis/pipelines/1/runs/471/signedartifactscontent?artifactName=invest-workbench-macos-latest&urlExpires=2021-02-16T18%3A23%3A52.4312404Z&urlSigningMethod=HMACV1&urlSignature=ADWtdCgV0fDVy4oh1o5gZyvHaHckMSnpG1eXXnYLNpI%3D

When I launch the application, the window launches great, and the model list is presented as expected.  I then launched `DelineateIt` from a prior run (this might not matter) and Mac OS needed to prompt me to confirm that I want to grant access to this directory.  The thing is, it was about 20-30 seconds before the OS permissions dialog was presented, and in that time there was nothing to indicate that the application was hanging.  Clearly the event loop is running, but the model inputs list wasn't showing up at all.

Maybe it'd be worth adding a spinner or other indicator to show that something is happening in the background on model launch?

**Comment by davemfish** [comment]
This sounds like a big deal and something we might want to fix for the alpha milestone. But can you elaborate a bit, @phargogh ? What directory are you being asked to grant access to? When you say "prior run", were you using the "Open" button or a "Recent Runs" button from the Home tab?

It should take virtually 0 seconds to open a model tab no matter how it was opened, so I'm curious what's actually happening in that 20-30 seconds. Also, I would not want OS permissions dialogs to be a normal part of the UX. Were you surprised by that, or is it a normal part of MacOS life?

**Comment by phargogh** [comment]
> What directory are you being asked to grant access to?

In this case it was `~/Documents`, which is where the workspace was located.  I believe this was autopopulated from a model run left over from either the development environment or another binary build.

> When you say "prior run", were you using the "Open" button or a "Recent Runs" button from the Home tab?

I was using the most recent of the "Recent Runs", which happened to be this run of "DelineateIt".

> Also, I would not want OS permissions dialogs to be a normal part of the UX. Were you surprised by that, or is it a normal part of MacOS life?

The only thing that was surprising was how long the dialog took to appear!  The rest is expected.  Apple has been gradually rolling out per-app permissions to specific parts of the filesystem in an effort to improve privacy protection.  Here's a screenshot showing the now-granted permissions in the Security Preferences pane:

![Screen Shot 2021-02-16 at 11 59 21 AM](https://user-images.githubusercontent.com/4583897/108115251-b4528900-704e-11eb-8e58-ebcc5686a927.png)

**Comment by phargogh** [comment]
I can't seem to get the permissions dialog to show up once again, but I'll screenshot it when I do.

**Comment by davemfish** [comment]
Thanks for the context! I wonder how are we managing this in Qt? I'm just guess here, but I think it would be `validation.py` rather than electron that is actually accessing that directory when the workspace input is populated. And so I'd expect we might encounter that with Qt as well?

And I guess we're unsure what causes the 20-30s delay. I've never seen that so I'm hoping it's related to that OS-dialog and not a separate issue.

**Comment by phargogh** [comment]
> I wonder how are we managing this in Qt?

Yeah, we aren't explicitly doing anything about it in Qt!  And also, there's no dialog / permissinos request and `InVEST.app` is not present in System Preferences at all, which is very strange.

I strongly suspect the delay is the result of my old laptop burning at ~60% CPU from other causes.  @emlys have you noticed any serious delays with electron requesting access to ~/Documents?

**Comment by emlys** [comment]
@phargogh I haven't been able to trigger the permissions request even with a fresh install. But it's happened before without any noticeable delay.

**Comment by phargogh** [comment]
@emlys OK great, that's good to hear!  @davemfish maybe this should be a beta-or-later issue?

**Comment by davemfish** [comment]
Okay, great to hear that the delay was an isolated incident. I'd rather not apply a progress indicator because I think this operation (opening a model) should be instantaneous. So if we find more cases where it's not, I'd rather fix the underlying issue than add a progress indicator.

But, perhaps this is a use-case for a dreaded cursor spinner/indicator? It would show up in the rare case where there is a delay.

**Comment by davemfish** [comment]
It sounds like there's nothing to do here unless we experience the long-wait times again.
